dags/common/utils.py

```diff
@@ -248,14 +248,41 @@ def create_or_update_article(data):
     )
     token = os.getenv("BACKEND_TOKEN", "CHANGE_ME")
     headers = {"Content-Type": "application/json", "Authorization": f"Token {token}"}
+
     response = requests.post(
-        f"{backend_url}",
+        backend_url,
         data=json.dumps(data),
         headers=headers,
     )
+
     try:
         response.raise_for_status()
         return response.json()
+    except requests.HTTPError as e:
+        error_content = response.json()
+        logger.error(f"Error message: {error_content.get('message')}")
+        if "article_id" in error_content:
+            return update_article(data, backend_url, error_content, headers)
+        raise requests.HTTPError(
+            f"Create failed: {error_content.get('message')}", response=e.response
+        )
+
+
+def update_article(data, backend_url, error_content, headers):
+    article_id = error_content["article_id"]
```

Comment:
1. since we have a check for `article_id` https://github.com/cern-sis/workflows/pull/193/files#diff-caa66940a5e0989c3d3a1a153a6658ead25d65b7f12ccef0d0efb3d46c032cecR264, there is no need to pass the `error_content` around.  We can just pass the `article_id`
2. we can easily reconstruct the backend_url from the env vars
3. also the `headers`, we can create a function to use in both of them or we can just reconstruct them inside the function, no need to pass it around.

At the end the function could have only 2 params:

```python
def update_article(article_id, data):
....
```